### DISCUSSION TITLE
PWA

### DISCUSSION THREAD

**Discussion by FlorianDevP** [likeandscribe/frontpage#165] [Ideas]
I like adding link aggregator sites like [dev.to](https://dev.to) to my phone Home Screen. It would be great to also have those for frontpage.
Sadly I have never configured a PWA so I have no idea how to do that. If nobody else has time to look into this, I could start doing some research.

**Comment by tom-sherman** [comment]
Hey!

We have some far off future plans to build a native apps, would that solve the same problem for you? We're unlikely to implement a PWA in the current codebase, but another thing we're planning to do in the future is offer a stable AppView API which would allow for third party clients that could implement PWA behaviour.

I'm going to convert this issue to a discussion so we can continue the conversation there ☺️

**Reply by FlorianDevP** [reply]
A native App would be even better. Though for the time being a PWA could work for some users. Would you accept a Pr if a added PWA support?

**Reply by FlorianDevP** [reply]
Frontpage is using Next.Js, which supports the [Web App Manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest).
[Creating a PWA with Next.js](https://nextjs.org/docs/app/building-your-application/configuring/progressive-web-apps#creating-a-pwa-with-nextjs)
Just this Manifest would allow users (like me) to install the frontpage.fyi webpage on their devices as a app like experience.
Notifications is much more work, but that would fit much more with a native app.

**Reply by tom-sherman** [reply]
I don't think we'd accept PRs for this right now, mostly because we have limited time and resources and need to be really careful about the things we take on to maintain right now.

As mentioned before though, an open and stable API is on the roadmap and so this kind of client could exist in the ecosystem.

**Reply by FlorianDevP** [reply]
Here's an example manifest from [daily.dev](https://app.daily.dev). (Also build with Next.Js)
```json
{
    "name": "daily.dev",
    "short_name": "daily.dev",
    "icons": [
        {
            "src": "/android-chrome-192x192.png",
            "sizes": "192x192",
            "type": "image/png",
            "purpose": "any"
        },
        {
            "src": "/android-chrome-512x512.png",
            "sizes": "512x512",
            "type": "image/png",
            "purpose": "any"
        },
        {
            "src": "/maskable_icon.png",
            "sizes": "512x512",
            "type": "image/png",
            "purpose": "maskable"
        }
    ],
    "theme_color": "#151618",
    "background_color": "#151618",
    "start_url": "/?pwa=true",
    "display": "standalone",
    "orientation": "portrait"
}
```

There also seem to be some backwards compatibility metadata tags in daily.dev's html head.
```html
<meta name="theme-color" content="#0E1217" data-next-head="">
<meta name="apple-mobile-web-app-status-bar-style" content="#0E1217" data-next-head="">
<meta name="application-name" content="daily.dev" data-next-head="">
<meta name="apple-mobile-web-app-capable" content="yes" data-next-head="">
<meta name="apple-mobile-web-app-title" content="daily.dev" data-next-head="">
<meta name="mobile-web-app-capable" content="yes" data-next-head="">
<link rel="apple-touch-icon" sizes="180x180" href="/apple-touch-icon.png" data-next-head="">
<link rel="icon" type="image/png" sizes="32x32" href="/favicon-32x32.png" data-next-head="">
<link rel="icon" type="image/png" sizes="16x16" href="/favicon-16x16.png" data-next-head="">
<link rel="manifest" href="/manifest.json" data-next-head="">
```
This is mostly for older iOS and Safari versions so I think it's not that important.

**Reply by FlorianDevP** [reply]
> I don't think we'd accept PRs for this right now, mostly because we have limited time and resources and need to be really careful about the things we take on to maintain right now.
> 
> As mentioned before though, an open and stable API is on the roadmap and so this kind of client could exist in the ecosystem.

Ok, I also sadly do not see any way that I could add this manifest myself.
Maybe in some really hacky way, but I guess I'm just going to wait for the native app.

**Reply by codegod100** [reply]
>We have some far off future plans to build a native apps

what framework? react native?

**Reply by will-corrigan** [reply]
Haven't really got anything concrete, I'm working on learning some react native in the background. It'll likely be with expo/rn though yeah.

**Comment by tom-sherman** [comment]
@FlorianDevPhynix just wanted to update to say that we've now landed a basic manifest, just to improve the experience of using the installed app somewhat.

https://github.com/likeandscribe/frontpage/commit/2d69054b5437266d7c80d81a5847ad2d4b0c2f72

We're not planning to add PWA capabilities right now though (as mentioned we have a native app planned for this), but hopefully it'll allow you to at least install it to your home screen now.

**Reply by FlorianDevP** [reply]
This works perfectly. Thank you very much!
![image](https://github.com/user-attachments/assets/30a71124-3c2a-4107-b25c-6291fd9eba61)
